daq/acl_state_collector.py

```diff
@@ -59,11 +55,40 @@ def _get_port_rules_count(self, switch, port, acl_config, acl_samples):
                 break
 
             if not has_sample:
-                LOGGER.warning(
-                    'No ACL metric sample available for switch, port, ACL, rule:'
-                    '%s, %s, %s, %s', switch, port, acl_config._id, cookie_num)
+                error = (f'No ACL metric sample available for switch, port, ACL, rule: '
+                         f'{switch}, {port}, {acl_config._id}, {rule_config["description"]} '
+                         f'(cookie={cookie_num})')
+                errors.append(error)
+                LOGGER.error(error)
+
+        return acl_map
+
+    def _verify_port_acl_config(self, switch, port):
+        error_map = {'errors': []}
+        error_list = error_map['errors']
+
+        switch_config = self._switch_configs.get(switch)
+        if not switch_config:
+            error = f'Switch not defined in Faucet dps config: {switch}'
+            LOGGER.error(error)
+            error_list.append(error)
+            return None, error_map
+
+        port_config = switch_config.ports.get(port)
+        if not port_config:
+            error = f'Port not defined in Faucet dps config: {switch}, {port}'
+            LOGGER.error(error)
+            error_list.append(error)
+            return None, error_map
+
+        acls_config = port_config.acls_in
+        if not acls_config:
+            error = f'No ACLs applied to port: {switch}, {port}'
+            LOGGER.error(error)
+            error_list.append(error)
+            return None, error_map
 
-        return {'rules': rules_map}
+        return acls_config[0], None
```

Comment:
also check if len(acls_config) != 1 -- since if it's >1 then information will be lost which is likely confusing -- or, if that's not possible do an assert to check len == 1